mico-core/src/main/java/io/github/ust/mico/core/web/DeploymentController.java

```diff
@@ -136,31 +144,27 @@ private String buildImageAndWait(MicoService micoService) {
-        if (micoApplication.getDeploymentInfo() != null &&
-            micoApplication.getDeploymentInfo().getServiceDeploymentInfos() != null) {
-            Map<Long, MicoServiceDeploymentInfo> serviceDeploymentInfos = micoApplication.getDeploymentInfo().getServiceDeploymentInfos();
-            Optional<MicoServiceDeploymentInfo> micoServiceDeploymentInfoOptional = Optional.ofNullable(serviceDeploymentInfos.get(micoService.getId()));
-            if (micoServiceDeploymentInfoOptional.isPresent()) {
-                micoServiceDeploymentInfo = micoServiceDeploymentInfoOptional.get();
-                log.debug("Using deployment information for MICO Service '{}' in version '{}': {}",
-                    micoService.getShortName(), micoService.getVersion(), micoServiceDeploymentInfo.toString());
-            } else {
-                log.warn("MICO application '{}' in version '{}' doesn't have a service deployment information for service '{}' in version '{}' stored.",
-                    micoApplication.getShortName(), micoApplication.getShortName(), micoService.getShortName(), micoService.getVersion());
-            }
+        Optional<MicoServiceDeploymentInfoQueryResult> serviceDeploymentInfoQueryResultOptional = serviceDeploymentInfoRepository
+                .findByApplicationAndService(micoApplication.getShortName(), micoApplication.getVersion(), micoService.getShortName(), micoService.getVersion());
+        MicoServiceDeploymentInfo serviceDeploymentInfo = new MicoServiceDeploymentInfo();
+        if (serviceDeploymentInfoQueryResultOptional.isPresent()) {
+            MicoServiceDeploymentInfoQueryResult serviceDeploymentInfoQueryResult = serviceDeploymentInfoQueryResultOptional.get();
+            serviceDeploymentInfo = serviceDeploymentInfoQueryResult.getServiceDeploymentInfo();
+            log.debug("Using deployment information for MICO Service '{}' in version '{}': {}",
+                    micoService.getShortName(), micoService.getVersion(), serviceDeploymentInfo.toString());
         } else {
-            log.warn("MICO application '{}' in version '{}' doesn't have any service deployment information stored.",
-                micoApplication.getShortName(), micoApplication.getShortName());
+            log.warn("MICO application '{}' in version '{}' doesn't have a service deployment information for service '{}' in version '{}' stored.",
+                    micoApplication.getShortName(), micoApplication.getShortName(), micoService.getShortName(), micoService.getVersion());
         }
         log.info("Creating Kubernetes deployment for MicoService '{}' in version '{}'",
             micoService.getShortName(), micoService.getVersion());
         log.debug("Details of MicoService: {}", micoService.toString());
-        micoKubernetesClient.createMicoService(micoService, micoServiceDeploymentInfo);
+        micoKubernetesClient.createMicoService(micoService, serviceDeploymentInfo);
```

Comment:
The MICO service will be created regardless of the existence of the `serviceDeploymentInfo`. The service should only be created, if the deployment information exists.

Reply:
Same question as with the endpoint for adding services to applications. I think the best option would be to check whether the deployment information is available. If so, use it, otherwise, simply apply the default deployment information.

I opened an issue for that: #476.